### ISSUE TITLE
In a Node.js environment, throws a resource lock error.

### ISSUE THREAD

**Issue by KotoriMinami** [Nozbe/WatermelonDB#1705]
My basic code is as follows.

```javascript
import { mySchema } from './model/schema.js'

import SQLiteAdapter from "@nozbe/watermelondb/adapters/sqlite/index.js";

const adapter = new SQLiteAdapter({
    dbName: 'test',
    schema: mySchema,
    onSetUpError: error => {
        // Database failed to load -- offer the user to reload the app or log out
        console.log(error)
    }
})
```
At this point, running the code produces the following results.
![image](https://github.com/Nozbe/WatermelonDB/assets/15208772/e60bf29a-c2fa-4111-bee2-9092dcdae9e4)
How should it be handled?

**Comment by GitMurf** [comment]
@KotoriMinami I am having this exact same issue. It looks like from the stack trace in your image that you are on Windows? I am on Windows as well.

See https://github.com/Nozbe/WatermelonDB/issues/1810#issuecomment-2351020396 for details on my issues.

**Comment by KotoriMinami** [comment]
@GitMurf 
Yes, I encountered this issue on Windows, but it's completely fine on Mac. 
I resolved it by applying the PR's code locally on my machine, 
and it's been running for about a year without any significant issues.
I'm not sure if your issue is the same as mine, but you can give it a try and see if it solves your problem.

**Comment by GitMurf** [comment]
> @GitMurf 
> 
> Yes, I encountered this issue on Windows, but it's completely fine on Mac. 
> 
> I resolved it by applying the PR's code locally on my machine, 
> 
> and it's been running for about a year without any significant issues.
> 
> I'm not sure if your issue is the same as mine, but you can give it a try and see if it solves your problem.

Thanks! And to be clear, all it needs are these 3 lines added?!

https://github.com/Nozbe/WatermelonDB/commit/59f87f171550920beb2941cdf5f33a43577d2915

**Comment by KotoriMinami** [comment]
> > I resolved it by applying the PR's code locally on my machine,
> > and it's been running for about a year without any significant issues.
> 
> Thanks! And to be clear, all it needs are these 3 lines added from here: [59f87f1](https://github.com/Nozbe/WatermelonDB/commit/59f87f171550920beb2941cdf5f33a43577d2915) ?!!

Yes, that's exactly how I did it.

**Comment by lsycxyj** [comment]
It works! I notice that it's been more than a year and it took me long time to figure it out. When will this PR be merged?

> > I resolved it by applying the PR's code locally on my machine,
> > and it's been running for about a year without any significant issues.
> 
> Thanks! And to be clear, all it needs are these 3 lines added from here: [59f87f1](https://github.com/Nozbe/WatermelonDB/commit/59f87f171550920beb2941cdf5f33a43577d2915) ?!!

It works! I notice that it's been more than a year and it took me long time to figure it out. When will this PR be merged?
